canonical/ubuntu.com
# WIP Setup certification test reults flutter widget demo

## Done

The widget is now showing up, but there are still path issues that need fixing.

- I could not use the `versioned_static` functionality, because that requires matched hashes of the script which is not possible because we are importing a build, therefore it will not generate hashes. I am laoding it directly, so you can see it works by visiting e.g. `/static/js/third-party/certification/flutter.js`
- [ ] it doesn't seem to load the `main.dart.js` script, my guess is that the paths inside the flutter widget could be wrong, maybe @mz2 could see better with the source files, as I am working with the final build and that is all minified, pretty hard to read.
- [ ] There are also issues importing the detarball, not sure yet what's causing it.
- [x] I'll keep investigating one of the errors that says `_flutter` is not defined, despite successfully loading the `flutter.js` file

## QA
1. If you don't have it, install the dotrun snap `snap install dotrun`
2. Checkout this PR and pull the code.
3. Go to the root directory and `dotrun`, it should install all dependencies. The flutter widget dependencies are under `/static/js/third-party` as are mounting the build files, this is probably where they should go.
4. I set up a test endpoint to view the flutter widget. `/certified/tests`. A working implementation should show the flutter widget on this page. The template is `/templates/certified/tests.html`


## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-6183

## Screenshots

[If relevant, please include a screenshot.]


## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)
